lib/logmatcher.c

```diff
@@ -425,50 +446,34 @@ static gboolean
 log_matcher_pcre_re_match(LogMatcher *s, LogMessage *msg, gint value_handle, const gchar *value, gssize value_len)
 {
   LogMatcherPcreRe *self = (LogMatcherPcreRe *) s;
-  gint *matches;
-  gsize matches_size;
-  gint num_matches;
   gint rc;
+  gint *matches = g_alloca(self->matches_size * sizeof(gint));
```

Comment:
Would it be possible to do the allocation also in pcre compile phase ?

Reply:
the whole point is that we don't want to share this buffer, and alloca() allocates on the stack, so not really an expensive operation.

If you really wanted to allocate in advance, I would use per-thread allocations where the thread_id (small integer number queried via main_loop_worker_set_thread_id() ), can be used to select which thread state is to be used.